### PR TITLE
[MRG] Remove Python 2.7 support (& end Python 2 compatibility)

### LINKED ISSUE #318: Drop Python 2 once khmer 3.0 is released.
Since khmer is [dropping Python 2][0], no reason to support it here.

[0]: https://github.com/dib-lab/khmer/issues/1769

### LINKED ISSUE #1033: remove support for Python 2.x, require python 3.7 or later
This is the issue that @luizirber has been waiting for for years!!

TODO  --

Make necessary documentation (install docs, requirements docs) changes.
Adjust conda recipe
Remove python 2.7 builder on CI.
Remove all of the __future__ statements.
change/fix Enum import stuff

remove compat code, including
```
sourmash/_compat.py:PY2 = sys.version_info[0] == 2
sourmash/lca/command_index.py:    if sys.version_info < (3, ):
sourmash/signature.py:                if sys.version_info >= (3,):
```

### LINKED ISSUE #1043: clean up signature module after python2.7 is dropped
per comment in `sourmash/cli/signature/__init__.py`

```
Copy commands over from 'sourmash sig'.                                         

This can be removed once Python 2.7 is no longer supported, in favor of an      
'aliases' argument to add_subparser in ../sig/__init__.py.     
```

### PR DESCRIPTION
**Pull request by ctb**
Remove all of the compatibility code and version checking necessary to support Python 2.7. 🎉 

Fixes #1033 
Fixes #318 
Fixes https://github.com/dib-lab/sourmash/issues/1043

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by luizirber** [comment]
We can also check any place where `sourmash._compat` is imported and used and remove

**Comment by luizirber** [comment]
(and remove 2.7 from CI)

**Comment by ctb** [comment]
> We can also check any place where `sourmash._compat` is imported and used and remove

(this turns out to require some actual work :)

File: README.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ A quickstart tutorial [is available](https://sourmash.readthedocs.io/en/latest/t
 
 ### Requirements
 
-sourmash runs under both Python 2.7.x and Python 3.5+.  The base
+sourmash runs under Python 3.7 and later.  The base
```
Comment (luizirber) [inline README.md]: oooh, 3.7? Sounds good! =]

**Review by luizirber — APPROVED** [review]

**Pull request merged**